### DISCUSSION TITLE
Can't compile current master

### DISCUSSION THREAD

**Discussion by kybos** [muse-sequencer/muse#1152] [Team Posts]
I cleaned up everything, deleted build folder, still there is always an error:

[100%] Linking CXX executable muse3
libmuse_lv2host_module.so: undefined reference to `MusECore::MidNamMIDIName::read(MusECore::Xml&)'
collect2: error: ld returned 1 exit status

**Comment by terminator356** [comment]
My mistake, hang in there...

**Comment by terminator356** [comment]
I'm out of time for tonight. Sorry about that.
I believe you can fix that by adding a line for midnam_module
 in muse/CMakeLists.txt at around line 360 like this,
 if it works feel free to commit it:

      if(HAVE_GTK2)
            target_link_libraries(lv2host_module lv2_gtk2_support)
      endif(HAVE_GTK2)

      target_link_libraries(lv2host_module ${QT_LIBRARIES})

      target_link_libraries(lv2host_module midnam_module)

**Comment by kybos** [comment]
OK, fixed and merged in master.

There are now some new warnings when MusE is started (I guess it's nothing critical):

XML: unexpected EOF in comment
XML Parse Error at line 6622 col 1
XML: unexpected EOF in comment
XML Parse Error at line 6622 col 1
XML: unexpected EOF in comment
XML Parse Error at line 255 col 1
XML: unexpected EOF in comment
XML Parse Error at line 1440 col 1
XML: unexpected EOF in comment
XML Parse Error at line 4171 col 1
WARNING: unexpected EOF reading xml file at level 3, line 4171, <MidiInstrument><><Yamaha 9000pro>
WARNING: unexpected EOF reading xml file at level 3, line 4171, <MidiInstrument><><Yamaha 9000pro>
XML: unexpected EOF in comment
XML Parse Error at line 7854 col 1
WARNING: unexpected EOF reading xml file at level 2, line 7854, <MidiInstrument><><Yamaha Motif XS>
XML: unexpected EOF in comment
XML Parse Error at line 6675 col 1
WARNING: unexpected EOF reading xml file at level 2, line 6675, <MidiInstrument><><Yamaha MX49/MX61>
XML: unexpected EOF in comment
XML Parse Error at line 1864 col 1
WARNING: unexpected EOF reading xml file at level 3, line 1864, <MidiInstrument><><Yamaha S30/S80>
WARNING: unexpected EOF reading xml file at level 3, line 1864, <MidiInstrument><><Yamaha S30/S80>
XML: unexpected EOF in comment
XML Parse Error at line 694 col 1
WARNING: unexpected EOF reading xml file at level 3, line 694, <SysEx><><Local off>
WARNING: unexpected EOF reading xml file at level 3, line 694, <SysEx><><Local off>

**Comment by terminator356** [comment]
WTF? For Pete's sake why didn't that show up here?
That's our xml instrument definition files (*.idf) causing that.
Likely several of your instruments are messed up if you look closely.
I made a change to our xml code comment-handling () to allow declarations (<!DOCTYPE ... >)
 because such a line is found in midnam xml. It was required to fix that.
I'll take a look. Sorry about that. Just can't win...

**Comment by terminator356** [comment]
OK My bad. Those messages were there but I missed them.
Should be fixed now. Thanks for spotting it.
